File: Code/Java-code/Mechanic.java
public class Mechanic extends Person{
    private String id;
    private JobType job;

    private ServiceAppointment[] serviceAppointments;

    public Mechanic(String name, int phoneNumber, JobType job)
    {
        this.setName(name);
        this.setPhoneNumber(phoneNumber);
        this.job = job;
    }
    public String getId() {
        return this.id;
    }

    public void setId(String id) {
        this.id = id;
    }

    
	public void setJob(JobType job) {
		this.job = job;
	}

	public JobType getJob() {
		return this.job;
	}

    public void setServiceAppointments(ServiceAppointment[] serviceAppointments) {
        this.serviceAppointments = serviceAppointments;
    }
    
    public String viewCarDetails()
    {
        String temp ="";
        for (ServiceAppointment serviceAppointment : serviceAppointments) {
            temp += serviceAppointment.getCar()+" // ";
            
        }
        return temp;
    }

    public String toString() {
        return this.getJob()+" name is "+ this.getName()+ " and phonenumber is "+ this.getPhoneNumber()+". ID number is "+ this.id + 
        "\n   "+ this.viewCarDetails();
    }
}

File: Code/PHP-Code/src/Mechanic.php
<?php
require_once "autoload.php";
class Mechanic extends Person {

    private String $id;
    private JobType $job;

    //private array ServiceAppointment $serviceAppointment;
    private $serviceAppointments = array();

    public function __construct(String $name, int $phoneNumber, JobType $job)
    {
        $this->setName($name);
        $this->setPhoneNumber($phoneNumber);
        $this->job = $job;
    }

    public function getId(): string
    {
        return $this->id;
    }

    public function setId(string $id): void
    {
        $this->id = $id;
    }

    public function setJob(JobType $job): void
    {
        $this->job = $job;
    }

    public function getJob(): JobType
    {
        return $this->job;
    }

    public function setServiceAppointment(ServiceAppointment $serviceAppointment): void
    {
        $this->serviceAppointment = $serviceAppointment;
    }

    public function viewCarDetails(): String
    {
        $temp = "";
        foreach ($this->serviceAppointments as $serviceAppointment) {
            $temp .= $serviceAppointment->getCar() . " // ";

        }
        return $temp;
    }


    public function __toString(): string
    {
        return "\n".$this->job->name." name is ". $this->getName(). " and phonenumber is ".$this->getPhoneNumber().". ID number is ". $this->id."\n";
    }

}